Create 2025-03-05.md (#718)
solid/specification: meetings/2025-03-05.md

### Participation and Code of Conduct
* [Join the W3C Solid Community Group](https://www.w3.org/community/solid/join), [W3C Account Request](http://www.w3.org/accounts/request), [W3C Community Contributor License Agreement](https://www.w3.org/community/about/agreements/cla/).
* [Solid Code of Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md), [Positive Work Environment at W3C: Code of Ethics and Professional Conduct](https://www.w3.org/Consortium/cepc/)
* Operating principle for effective participation is to allow access across disabilities, across country borders, and across time. Feedback on tooling and meeting timing is welcome.
* If this is your first time, welcome! please introduce yourself.

### Scribes

* elf Pavlik

## Topics

### PRACTITIONERS MEETUP ON TRANSPORTATION
* JZ: Thursday March 6, 15-17:00 UTC https://meet.jit.si/solid_practioners will focus on Solid and Transportation with talks from Pietr Colpaert and four others working on decentralized transportation.  All invited.

### Connecting a Wallet with a Pod

> * MdJ: Inrupt, KIT and SURF are all experimenting with the connection between pods and wallets. Let's compare notes! https://docs.inrupt.com/data-wallet/patterns-and-flows/
> * MdJ: We are doing it with the Dutch Univeristy. We try to share VCs from my wallet. Instead of it being self contained it refers to a document on the pod. The question is how the signature works then. Does the document contain signature. How does it work with the Inrupt wallet. Can you share a document in the same way as you would share a VC?

Reviewer: ```suggestion
* MdJ: Inrupt, KIT, and SURF are all experimenting with the connection between pods and wallets. Let's compare notes! https://docs.inrupt.com/data-wallet/patterns-and-flows/
* MdJ: We are doing it with the Dutch Univeristy. We try to share VCs from my wallet. Instead of it being self contained, it refers to a document on the pod. The question is,  how does the signature work then? Does the document contain signature? How does it work with the Inrupt wallet? Can you share a document in the same way as you would share a VC?
```

> * HZ: Yes you can. It is an API to make the interaction easier. There is an inbox, kind of like a notifications. You can share document, in the mobile app there is a QR code if someone wants to use it. The request is being made to get the document.

Reviewer: ```suggestion
* HZ: Yes, you can. It is an API to make the interaction easier. There is an inbox, kind of like notifications. You can share a document. In the mobile app, there is a QR code, if someone wants to use it. The request is made to get the document.
```

* MdJ: Is anything signed?
* HZ: ...

> * MdJ: Are you aware of any extension of OIDC4VP that would allow it.
> * HZ: No

Reviewer: ```suggestion
* MdJ: Are you aware of any extension of OIDC4VP that would allow it?
* HZ: No.
```

> * HZ: It is not that when Inrupt donated it to OWF and wanted it to be the story. It wanted cross pollination and the API to be inproved. We don't care about the wallet but about the storage. It is open source code licensed under Apache.
> * eP: Yesterday I joined the credentials CG, where someone from Octa and Sam from Google were presenting the digital credentials API, and discussed the differences with FedCM. There are multiple protocols, like OpenID4VP. It uses FIDO etc. and QR codes. What is the overlap between that and the wallets you are discussing now? OM th

Reviewer: ```suggestion
* HZ: It is not that when Inrupt donated it to OWF and wanted it to be the story. It wanted cross pollination and the API to be improved. We don't care about the wallet, but about the storage. It is open source code, licensed under Apache.
* eP:, Yesterday I joined the credentials CG, where someone from Octa and Sam from Google were presenting the digital credentials API, and discussed the differences with FedCM. There are multiple protocols, like OpenID4VP. It uses FIDO, etc., and QR codes. What is the overlap between that and the wallets you are discussing now?
```

> * MdJ: I think it is pretty related. AFAIK Credentials CG is about interfacing wallets with the web. You could have a boarding pass or driver's license on your phone, and CG describes how it could be shared in the browser to the website. You share it either with the QR code or with NFC to some device, similar to payments. They also add how to present it via JS API. What I'm doing is about wallets, but not necessarly adding interaces and web page but between them and storage servers.
> * HZ: The wallet API, the backend is documented and open. Did you take a look at it and would it be usefull in your project? It could be useful to have multiple implementations. There is also ane from South American company which is doing attachments on VCs.

Reviewer: ```suggestion
* MdJ: I think it is pretty related. AFAIK Credentials CG is about interfacing wallets with the web. You could have a boarding pass or driver's license on your phone, and CG describes how it could be shared in the browser to the website. You share it either with the QR code or with NFC to some device, similar to payments. They also add how to present it via JS API. What I'm doing is about wallets, but not necessarly adding interfaces and web page, but between them and storage servers.
* HZ: The wallet API, the backend, is documented and open. Did you take a look at it, and would it be useful in your project? It could be useful to have multiple implementations. There is also ane from South American company which is doing attachments on VCs.
```

* eP: In the case you're describing, connecting to storage, how is the user involved?

> * MdJ: The user would not see much of a difference between sharing a credential on their device vs sharing one in their storage. You could have them in both location. It could be transparent to the user. The Verifiers would not receive the whole thing in the QR code or NFC interaction. They could receive a read capability for example. 
> * eP: so where the wallet is makes the main difference, device vs pod

Reviewer: ```suggestion
* MdJ: The user would not see much of a difference between sharing a credential on their device vs sharing one in their storage. You could have them in both location. It could be transparent to the user. The Verifiers would not receive the whole thing in the QR code or NFC interaction. They could receive a read capability, for example. 
* eP: So where the wallet is makes the main difference, device vs pod.
```

* MdJ: ... There might be a big overlap with organizational wallets. What I will build will be a combination of a mobile app with pod.

> * eP: It would be interesting to see how it plays in local-first setup. Where VCs would be replicaterd.
> * HZ: Do you know that the open wallet Inrupt donated to OWF is white labeled?
> * MdJ: I'm using API but not building the app.
> * HZ: We want to make it pluggable on the backend. On the frontend you may want to change what is being displayed etc.

Reviewer: ```suggestion
* eP: It would be interesting to see how it plays in local-first setup. Where VCs would be replicated.
* HZ: Do you know whether the open wallet Inrupt donated to OWF is white labeled?
* MdJ: I'm using API but not building the app.
* HZ: We want to make it pluggable on the backend. On the frontend, you may want to change what is being displayed, etc.
```

### PATCH

> * MdJ: The way I see it, https://github.com/solid/specification/issues/715 sounds like it would not be a minor bugfix, so I think it would be more a task for LWS WG, not for Solid CG.
> * MdJ: Regarding https://github.com/solid/specification/issues/715 I would initially say it sounds like a minor bugfix, but after reading about how difficult this would be for implementors, given the apparent unavailability of off-the-shelf parsers that could deal with this, I don't think it's within remit for Solid CG to merge a PR like this, so again, this feels to me like something we should leave for LWS WG to work on.

Reviewer: ```suggestion
* MdJ: The way I see it, https://github.com/solid/specification/issues/715 sounds like it would not be a minor bug fix, so I think it would be more a task for LWS WG, not for Solid CG.
* MdJ: Regarding https://github.com/solid/specification/issues/715, I would initially say it sounds like a minor bug fix, but after reading about how difficult this would be for implementers, given the apparent unavailability of off-the-shelf parsers that could deal with this, I don't think it's within remit for Solid CG to merge a PR like this, so again, this feels to me like something we should leave for LWS WG to work on.
```

> * MdJ: Regarding textual changes, yes, I think they are necessary and I think we can work on them in GitHub PRs and discuss them in Solid  CG calls. In particular it seems there is one mention of blank nodes in a place where blank nodes are not allowed, we should probably just remove that mention: https://github.com/solid/specification/pull/716
> * JW: I see a few issues conflated. The discussion Ruben had, about no off the shelf parsers. The issue is that currently N3 doesn't define how blank notes scoping works. It is undefined if two blank nodes with the same label in different graphs are co-reffering or distinct.

Reviewer: ```suggestion
* MdJ: Regarding textual changes, yes, I think they are necessary. I think we can work on them in GitHub PRs, and discuss them in Solid CG calls. In particular, it seems there is one mention of blank nodes in a place where blank nodes are not allowed, we should probably just remove that mention: https://github.com/solid/specification/pull/716
* JW: I see a few issues being conflated. The discussion Ruben had, about no off-the-shelf parsers. The issue is that currently N3 doesn't define how blank nodes are scoped. It is undefined whether two blank nodes with the same label in different graphs are coreferences or distinct.
```

> * JW: For N3 patch this isn't much of an issue. We don't propose that deletions are allowed to refer to blank node. Besides the patch identifier itself there are no blank nodes in the document. We can propose a compromise that bypasses all of the issue mentioned by Ruben.
> * JW: Dorthe have commented that there is an interest in N3 WG that would resolve that issue with blank node scoping. We can procede with the insertion and be aware with the pitfalls in current parsers.

Reviewer: ```suggestion
* JW: For N3 Patch, this isn't much of an issue. We don't propose that deletions are allowed to refer to blank nodes. Besides the patch identifier itself, there are no blank nodes in the document. We can propose a compromise that bypasses all of the issue mentioned by Ruben.
* JW: Dörthe has commented that there is an interest in N3 WG that would resolve that issue of blank node scoping. We can proceed with the insertion and be aware of the pitfalls in current parsers.
```

> * MdJ: Have anyone implemented it yet?
> * JW: NSS always allowed it, CSS forbids it. CSS uses N3 which can parse those blank nodes, there is just one line in CSS that checks if the blank nodes exist. I understand Ruben's argument but I see it as academic precission.
> * MdJ: If NSS already does it, if CSS doesn't do it we can do it in Pivot. ESS doesn't do N3 patch at all.

Reviewer: ```suggestion
* MdJ: Has anyone implemented it yet?
* JW: NSS always allowed it; CSS forbids it. CSS uses N3 which can parse those blank nodes; there is just one line in CSS that checks whether the blank nodes exist. I understand Ruben's argument, but I see it as academic precision.
* MdJ: If NSS already does it, even if CSS doesn't do it, we can do it in Pivot. ESS doesn't do N3 Patch at all.
```

* eP: Do you know why ESS doesn't support it?

> * MdJ: The whole patching is in the air. When the switch away from SPARQL update happended, it was known that WG will be resolving that. So this wasn't a priority to implement what was proposed in CG.
> * RG: Why is it that we put N3 patch as specification inside solid protocol spec instead having it as a normative reference. Jesse can you make that comment about planned change in the PR as a note?

Reviewer: ```suggestion
* MdJ: The whole area of patching is in the air. When the switch away from SPARQL Update happened, it was known that WG would be resolving that, so it wasn't a priority to implement what was proposed in CG.
* RG: Why is it that we put N3 Patch as a specification inside the Solid Protocol spec instead of having it as a normative reference? Jesse, can you make that comment about planned change in the PR as a note?
```

> * MdJ: Why don't we separate it from the protocol spec. It seems to match suggestion that there will be a N3 WG.
> * RG: Would they also take responsibility of N3 patch?
> * MdJ: That would also help to keep Solid CG on topic. If there is an N3 CG we could move the patch there. In the protocol we can make a chance to refer to it as a normative reference.

Reviewer: ```suggestion
* MdJ: Why don't we separate it from the protocol spec? It seems to match the suggestion that there will be an N3 WG.
* RG: Would they also take responsibility for N3 Patch?
* MdJ: That would also help to keep Solid CG on topic. If there is an N3 CG, we could move N3 Patch there. In the protocol, we can make a choice to make normative reference to it.
```

> * JW: I don't think N3 CG would be interested in patch to be honset. It is describing how to modify resources. There is no other write type interface in N3 spec at this moment. It defines semantic, inference and query. It doesn't describe any mechanism to update.
> * RG: For now we just make a change within the Solid Spec, we take it in Solid CG and take it into a separate document. Once we have N3 Patch document we can ask them if they want to do it. Or we can hand it to LWS.

Reviewer: ```suggestion
* JW: I don't think N3 CG would be interested in patch to be honest. It is describing how to modify resources. There is no other write type interface in the N3 spec at this moment. It defines semantics, inference, and query. It doesn't describe any mechanism for update.
* RG: For now, we just make a change within the Solid Spec, we take it to Solid CG, and take it into a separate document. Once we have N3 Patch document, we can ask N3 WG if they want to do it. Or we can hand it to LWS WG.
```

> * eP: +1 to extract it as a separate draft report
> * JW: That sounds reasonable
> * MdJ: can we make that change in Pivot
> * JW: I don't have time at this moment. I can look into it at some poiont. 
> * MdJ: Since NSS already does it, we should also do it in Pivot

Reviewer: ```suggestion
* eP: +1 to extract it as a separate draft report.
* JW: That sounds reasonable.
* MdJ: Can we make that change in Pivot.
* JW: I don't have time at this moment. I can look into it at some point. 
* MdJ: Since NSS already does it, we should also do it in Pivot.
```

* HZ: Sounds good.
* MdJ: We are just incubating here as CG ...
* eP: Are we going to merge that PR?

> * RG: I can do it but not before April 1st or 3rd
> * eP: So we merge the PR first and later we extract it?
> * RG: Yes, merge PR first and do extraction later.
> * RG: I can do the mechanical extraction but I can't be the editor since I'm still new to RDF
> * JW: I can help with RDF bits

Reviewer: ```suggestion
* RG: I can do it but not before April 1st or 3rd.
* eP: So we merge the PR first and later we extract it?
* RG: Yes, merge PR first and do extraction later.
* RG: I can do the mechanical extraction, but I can't be the editor, since I'm still new to RDF.
* JW: I can help with RDF bits.
```

### solidcommunity.net

> * JW: We removed all ssh access to it to tighten up security. We will setup a separate droplet for more experimental deployments.
> * MdJ: Will we be able to see the logs?
> * JW: I can do it, I would like to setup a service to offload logs from the server.
> * HZ: What the plans are to replace Solid Team admin group, which is just you at this moment. For the solidcommunity.net

Reviewer: ```suggestion
* JW: We removed all `ssh` access to it to tighten up security. We will set up a separate droplet for more experimental deployments.
* MdJ: Will we be able to see the logs?
* JW: I can do it. I would like to set up a service to offload logs from the server.
* HZ: What are the plans to replace Solid Team admin group, which is just you at this moment? For the solidcommunity.net?
```

> * JW: We will have ODI staff maintaining the server. Configuration changes will be changed via PRs to Pivot.
> * Theo: I told you at FOSDEM that indiehosters could make a partnership to help with hosting.
> * JW: Where are they located?
> * Theo: I think mostly France
> * JW: What do they want to do? What's their company end goal?

Reviewer: ```suggestion
* JW: We will have ODI staff maintaining the server. Configuration changes will be changed via PRs to Pivot.
* Theo: I told you at FOSDEM that indiehosters could make a partnership to help with hosting.
* JW: Where are they located?
* Theo: I think mostly France.
* JW: What do they want to do? What's their company end goal?
```

> * Theo: They have servers and would be happy to provide them for projects like Solid. Also the maintenanc of the server.
> * MdJ: https://www.wired.com/2014/11/indie-hosters/
> * JW: Hosting is not a big cost at this moment at Digital Ocean. Maintenance is a bigger cost. There is a risk in trusting an external company. For now it is a great offer but we don't have the capacity to effectively take up on it. I would suggesting to put it on the back burner for now.

Reviewer: ```suggestion
* Theo: They have servers and would be happy to provide them for projects like Solid. Also the maintenance of the server.
* MdJ: https://www.wired.com/2014/11/indie-hosters/
* JW: Hosting at Digital Ocean is not a big cost at the moment. Maintenance is a bigger cost. There is a risk in trusting an external company. For now, it is a great offer, but we don't have the capacity to effectively take up on it. I would suggest we put it on the back burner for now.
```

> * RG: Would there be a possibility to see my data on the pod on a different port?
> * JW: I shut down those additional ports, I understand they were showing the same data.
> * RG: The idea was that the standard port was running the stable version, and other ports allowed to test other configurations.
> * JW: There will be a separate droplet and a different domain. Ideally we will just wipe it every 24h to make clear we jsut use it for testing.

Reviewer: ```suggestion
* RG: Would there be a possibility to see my data on the pod on a different port?
* JW: I shut down those additional ports. I understand they were showing the same data.
* RG: The idea was that the standard port would be running the stable version, and other ports would be allowed to test other configurations.
* JW: There will be a separate droplet and a different domain. Ideally, we will wipe it every 24h, to make clear we just use it for testing.
```

> * RG: It is useful to have place to test things.
> * JW: What prevents you from doing experiments on localhost?
> * RG: You do that first but sometimes you need to do implement things on remote server. For example notifications. 
> * eP: +1 to separate it
> * MdJ: I have a droplet which I just swhich on and off paying per hour of using it.
> * RG: I hope that we will keep a common account. We want an infrastructure that all contributors could use without having to pay for their own servers. I find it a very convienient to have something like Alain's previous setup.

Reviewer: ```suggestion
* RG: It is useful to have a place to test things.
* JW: What prevents you from experimenting on localhost?
* RG: You do that first, but sometimes you need to implement things on a remote server. For example, notifications. 
* eP: +1 to separate it.
* MdJ: I have a droplet which I just switch on and off, paying per hour of using it.
* RG: I hope that we will keep a common account. We want an infrastructure that all contributors could use without having to pay for their own servers. I find it very convenient to have something like Alain's previous setup.
```
